internal/compiler/widgets/fluent/std-widgets.slint

```diff
@@ -332,12 +358,43 @@ export TabImpl := Rectangle {
         x: (parent.width - width) / 2;
     }
 
-    fs := FocusScope {}
+    Rectangle { // Focus Rectangle
+        border-width: enabled && has-focus ? 1px : 0px;
+        border-color: Palette.black;
+    }
 }
 
 export TabBarImpl := HorizontalLayout {
+    property<int> current; // The currently selected tab
+    property<int> current-focused: fs.has-focus ? i-focused-tab : -1; // The currently focused tab
+    property<int> num-tabs; // The total number of tabs
+
+    property<int> i-focused-tab: 0; // The currently focused tab
```

Comment:
For private property, you can put them in a child,  like in the `fs` for example.